### PR TITLE
Refactor time limit handling in DeliveryResponse and DeliveryService

### LINKED ISSUE #8: 4xx 抽出で使う timelimit 入力の安全性（サニタイズとバインド）
(GPT-5 によるコードレビューの結果、問題点をIssueに落としてもらったもののひとつです)

## 8) 4xx 抽出で使う timelimit 入力の安全性（サニタイズとバインド）
- 重要度: 高
- 優先度: 中
- 背景 / 現状の問題:
  - `time_limit_string`（例: `"72:00:00"`）を、そのまま SQL 文字列に埋め込んでいます。形式が不正な文字列を渡すと、SQL インジェクションになる危険があります。
- 目的:
  - 不正な入力を確実に拒否し、SQL にはプレースホルダ（バインド変数）で値を渡す。
- 対応方針:
  - 許可する形式を厳格に限定します（`HH:MM:SS` のみ）。
  - 可能なところは Ruby 側で期限境界（比較時刻）を算出し、SQL には日時やフラグなどの値をバインドして渡します。
- 変更範囲:
  - `app/models/delivery_response.rb`（生の文字列埋め込みを排除）、呼び出し側の引数検証（Rake タスク側）。
- 受け入れ基準:
  - 不正形式を与えると実行されず、明確なエラーが返る。インジェクション用のテストが失敗する（= 防げている）。

### PR DESCRIPTION
**Pull request by hiroaki**
#8 の対応です。
主な修正点は値を SQL に直接埋め込んでいる箇所の修正です。
またこの際ついでなので、引数のところの仕様も見直し、またSQLもGPT に見直してもらいました。

---

This pull request refactors and improves the handling of time limits for retrying mail deliveries, focusing on clearer parameter handling, stricter input validation, and improved test coverage for edge cases. The main change is to use seconds (as integer or ActiveSupport::Duration) instead of a string for time limits in the core logic, while still allowing string input at the service layer with proper parsing and validation.

**Core logic and validation improvements:**

* Refactored `DeliveryResponse.last_status_4xx_within_time_limit` to accept time limits in seconds (integer or ActiveSupport::Duration) instead of a string, with strict validation to raise an error for non-positive values. The SQL logic was updated for clarity and correctness.
* Added a new `parse_timelimit_seconds` method in `DeliveryService` to validate and convert 'HH:MM:SS' strings to seconds, raising an error for invalid formats or non-positive values.

**Service layer adjustments:**

* Updated `prepare_to_retry_for_session` in `DeliveryService` to use the new parsing method, ensuring time limits are consistently validated and converted before use. Also defined a `DEFAULT_TIME_LIMIT` constant for default values. [[1]](diffhunk://#diff-66b409fa39d6a56d6d6c8b6a236071f7262eb1cf7d72b50c748c698974256d15R33-R34) [[2]](diffhunk://#diff-66b409fa39d6a56d6d6c8b6a236071f7262eb1cf7d72b50c748c698974256d15L68-R72)

**Test enhancements:**

* Expanded model specs for `DeliveryResponse` to cover input validation, edge cases (such as nil, invalid strings, floats, and boundary values), and correct filtering logic. [[1]](diffhunk://#diff-72f99939b44b24b3b5109255daa5ecdac475294f03147d39a679003378911dafR12-R17) [[2]](diffhunk://#diff-72f99939b44b24b3b5109255daa5ecdac475294f03147d39a679003378911dafL54-R60) [[3]](diffhunk://#diff-72f99939b44b24b3b5109255daa5ecdac475294f03147d39a679003378911dafL68-R90) [[4]](diffhunk://#diff-72f99939b44b24b3b5109255daa5ecdac475294f03147d39a679003378911dafR99-R138)
* Added service specs to verify that invalid or zero time limit strings raise `ArgumentError` in `prepare_to_retry_for_session`.

### CODE REVIEW

File: app/models/delivery_response.rb [hunk 1]
```diff
@@ -1,58 +1,32 @@
...
-  #
-  # WARN:
-  #   - SQL の中の HAVING... のところで、 #strftime を使っています。
-  #     これについては #to_fs(:db) を使わないでください。 UTC の時刻になってしまいます。
-  def self.last_status_4xx_within_time_limit(time_limit_string = '72:00:00') # 初期値 72時間=3日間
-    DeliveryResponse.find_by_sql(<<-SQL2
-      SELECT * FROM delivery_responses AS T0
+  # 返される DeliveryResponse のリストにひもづく各 MailQueue は、
+  # まだ再送信猶予時間内にあり、再送信を試みる候補となります。
+  def self.last_status_4xx_within_time_limit(time_limit_seconds = 72.hours)
```
[the remark this answers is not in the document]
Comment (hiroaki) [reply app/models/delivery_response.rb:11]: 「秒数（整数）」または ActiveSupport::Duration を受ける設計に統一し、誤って小数を渡して暗黙に切り捨てられる事故を防ぎたいので、 Integer のチェックでよいです
